manager/allocator/network.go

```diff
@@ -1089,7 +1089,11 @@ func updateTaskStatus(t *api.Task, newStatus api.TaskState, message string) {
 
 // IsIngressNetwork returns whether the passed network is an ingress network.
 func IsIngressNetwork(nw *api.Network) bool {
-	if nw.Spec.Ingress {
+	cnmSpec := nw.Spec.GetCNM()
+	if cnmSpec == nil {
+		return false
+	}
```

Comment:
I was just thinking along the lines of a `GetCNMSpec(spec)` helper which returns a CNMSpec, either directly the result of `nw.Spec.GetCNM()` or a freshly fabricated one from the legacy fields. Except that doesn't handle the right back case very elegantly, I'll have to have a think/play with that.